deepchecks/nlp/context.py

```diff
@@ -0,0 +1,417 @@
+                 y_pred_train: TTextPred = None,
+                 y_proba_train: TTextProba = None,
+                 validate_data_on_predict: bool = True):
+        """Initialize dummy model."""
+        if train is not None:
+            if train.name is None:
+                train.name = 'train'
+        if test is not None:
+            if test.name is None:
+                test.name = 'test'
+
+        predictions = {}
+        probas = {}
+
+        if ((y_proba_train is not None) or (y_proba_test is not None)) and \
+                (train.task_type == TaskType.TOKEN_CLASSIFICATION):
+            raise DeepchecksNotSupportedError('For token classification probabilities should be part of the token'
+                                              ' prediction annotation and not passed to the proba argument')
+
+        for dataset, y_pred, y_proba in zip([train, test],
+                                            [y_pred_train, y_pred_test],
+                                            [y_proba_train, y_proba_test]):
+            if dataset is not None:
+                if y_pred is not None:
+                    self._validate_prediction(dataset, y_pred)
+                if y_proba is not None:
+                    self._validate_proba(dataset, y_proba)
+
+                if dataset.task_type == TaskType.TEXT_CLASSIFICATION:
+                    if (y_pred is None) and (y_proba is not None) and \
+                            (dataset.task_type == TaskType.TEXT_CLASSIFICATION):
+                        if dataset.is_multilabel:
+                            y_pred = np.array(y_proba) > 0.5
+                        else:
+                            y_pred = np.argmax(np.array(y_proba), axis=-1)
+                    else:
+                        y_pred = np.array(y_pred)
+                        if len(y_pred.shape) > 1 and y_pred.shape[1] == 1:
+                            y_pred = y_pred[:, 0]
+                        ensure_predictions_shape(y_pred, dataset.text)
+
+                y_pred_dict = dict(zip(dataset.index, y_pred))
+                predictions[dataset.name] = y_pred_dict
+                if y_proba is not None:
+                    ensure_predictions_proba(y_proba, y_pred)
+                    y_proba_dict = dict(zip(dataset.index, y_proba))
+                    probas[dataset.name] = y_proba_dict
+
+        self.predictions = predictions if predictions else None
+        self.probas = probas if probas else None
+        self.validate_data_on_predict = validate_data_on_predict
+
+        if self.predictions is not None:
+            self.predict = self._predict
+
+        if self.probas is not None:
+            self.predict_proba = self._predict_proba
+
+    def _predict(self, data: TextData) -> TTextPred:
+        """Predict on given data by the data indexes."""
+        return list(itemgetter(*data.index)(self.predictions[data.name]))  # pylint: disable=unsubscriptable-object
```

Comment:
How would you do that differently? I can cast the predictions to a numpy array, but it seems odd because they can be a sequence of sequence of tuples in case of token classification, which is not a format that quite fits numpy arrays.

Reply:
I like the item getter

Reply:
why not have same behavior as the one in tabular?

Reply:
I do like this itemgetter

Reply:
Cool, then it's settled